src/map.cpp

```diff
@@ -7035,6 +7035,25 @@ void map::rad_scorch( const tripoint &p, int time_since_last_actualize )
     }
 }
 
+void map::decay_gibs( const tripoint &p, int time_since_last_actualize )
+{
+    for( auto &pr : field_at( p ) ) {
+        auto &fd = pr.second;
+        if( !fd.decays_on_actualize() ) {
+            continue;
+        }
+
+        const int added_age = 2 * time_since_last_actualize / rng( 2, 4 );
+        fd.setFieldAge( fd.getFieldAge() + added_age );
```

Comment:
Not related to the PR directly, but anyways. The idiom `setFieldAge( getFieldAge() ... )` dominates in the code (`setFieldAge( something absolute )` is a rare thing). It would be really nice to introduce a `mod_age()` function.